src/app/home/home.component.ts

```diff
@@ -125,6 +127,24 @@ export class HomeComponent implements OnInit {
         this.currentTime = simulatedDate;
       }
     }
+
+    // start the redirect 12 hours in advance of CC open:
+    const startRedirectingToCCAt = new Date('2023-11-28T00:00:00+00:00');
+
+    // end the redirect exactly at the time CC closes.
+    if (
+      !queryParams.hasOwnProperty('noredirect') &&
+      this.currentTime >= startRedirectingToCCAt &&
+      this.currentTime < CCCloseDate &&
+      isPlatformBrowser(this.platformId)) {
+      this.router.navigate(
+        ['/christmas-challenge-2023'],
+        {
+          replaceUrl: true, // As we are redirecting imediatly it would be confusing to leave a page the user hasn't seen in their history.
```

Comment:
```suggestion
          replaceUrl: true, // As we are redirecting immediately it would be confusing to leave a page the user hasn't seen in their history.
```